src/opencl_common.h

```diff
@@ -274,6 +298,158 @@ void opencl_process_event(void);
 		} \
 	} while (0)
 
+/*
+ * Use this macro for OpenCL buffer creation.
+ * Later, it can release all buffers at once
+*/
+#define CLCREATEBUFFERx(_flags, _size, error_level, buffer_list, buffer_size) ({    \
+    cl_mem __buffer;                                                                \
```

Comment:
I need a better name (`CLCREATEBUFFER_ADD_LIST`)

Reply:
Keeping it short-ish it could be `CLCREATEBUF_LIST`. It could even be `CLCREATEBUFFERS` vs `CLCREATEBUFFER`...

Perhaps some of these things should be real functions in `opencl_common.c` instead of function-like macros.

Reply:
Which ones?

Reply:
Different perspective: why are any of these macros at all? Shouldn't all of them be functions?

Reply:
Right!

Reply:
> Keeping it short-ish it could be `CLCREATEBUF_LIST`. It could even be `CLCREATEBUFFERS` vs `CLCREATEBUFFER`...

I misunderstood your functions/macros yesterday. I thought the one now called `opencl_create_buf_list()` would be able to allocate several/all buffers for a format, in one call. Now I see `_list` refers to "keeping track".

Given that, I'm not sure the name need to reflect that at all (just like `malloc()` keeps lists and whatnot under the hood but it's not called `malloc_list()`). Maybe we should *only* export a function that does keep track, yet just call it `opencl_create_buf()`.